zomes/hc_zome_trust_atom/src/trust_atom.rs

```diff
@@ -125,102 +137,112 @@ fn trust_atom_link_tag(link_direction: &LinkDirection, mut chunks: Vec<&str>) ->
   let content = chunks.remove(0);
   link_tag_bytes.extend_from_slice(content.as_bytes());
 
-  for chunk in &chunks {
+  for chunk in chunks {
     link_tag_bytes.extend_from_slice(&UNICODE_NUL_BYTES);
     link_tag_bytes.extend_from_slice(chunk.as_bytes());
   }
+  link_tag_bytes.extend_from_slice(&UNICODE_NUL_BYTES);
+  for digit in bucket {
+    link_tag_bytes.push(digit);
+  }
 
   LinkTag(link_tag_bytes)
 }
 
-pub fn query_mine(
-  target: Option<EntryHash>,
-  content_full: Option<String>,
-  content_starts_with: Option<String>,
-  min_value: Option<String>,
-) -> ExternResult<Vec<TrustAtom>> {
-  let agent_address: EntryHash = agent_info()?.agent_initial_pubkey.into();
+fn gen_bucket() -> ExternResult<Vec<u8>> {
+  let bucket = random_bytes(9)?.into_vec();
```

Comment:
this will be 9 rand bytes, but not the ascii bytes representing 0-9 (could be any).  Checking on the test, should catch this...